packages/bson-bench/src/task.ts

```diff
@@ -174,31 +178,39 @@ export class Task {
 
     // install required modules before running child process as new Node processes need to know that
     // it exists before they can require it.
-    const pack = new Package(this.benchmark.library);
-    if (!pack.check()) await pack.install();
-    // spawn child process
-    const child = fork(`${__dirname}/base`, {
-      stdio: ['inherit', 'inherit', 'inherit', 'ipc'],
-      serialization: 'advanced'
-    });
-    child.send({ type: 'runBenchmark', benchmark: this.benchmark });
-    this.children.push(child);
-
-    // listen for results or error
-    const resultOrError: ResultMessage | ErrorMessage = (await once(child, 'message'))[0];
-
-    // wait for child to close
-    await once(child, 'exit');
-
-    this.hasRun = true;
-    switch (resultOrError.type) {
-      case 'returnResult':
-        this.result = resultOrError.result;
-        return resultOrError.result;
-      case 'returnError':
-        throw resultOrError.error;
-      default:
-        throw new Error('Unexpected result returned from child process');
+    if (!(await exists(Task.packageInstallLocation))) {
+      await mkdir(Task.packageInstallLocation);
+    }
+
+    try {
+      const pack = new Package(this.benchmark.library, Task.packageInstallLocation);
+      if (!pack.check()) await pack.install();
+      // spawn child process
+      const child = fork(`${__dirname}/base`, {
+        stdio: ['inherit', 'inherit', 'inherit', 'ipc'],
+        serialization: 'advanced'
+      });
+      child.send({ type: 'runBenchmark', benchmark: this.benchmark });
+      this.children.push(child);
+
+      // listen for results or error
+      const resultOrError: ResultMessage | ErrorMessage = (await once(child, 'message'))[0];
+
+      // wait for child to close
+      await once(child, 'exit');
```

Comment:
We're awaiting message on line 197 which means we could miss the exit event, we have to make sure the listeners are added synchronously to the child object or this will block. 

Of course, this is only if something is wrong, but a hang can be hard to debug.